python-package/lightgbm/sklearn.py

```diff
@@ -1008,7 +1008,7 @@ def fit(self, X, y,
                     callbacks=callbacks, init_model=init_model)
         return self
 
-    _base_doc = LGBMModel.fit.__doc__.replace("self : LGBMModel", "self : LGBMClassifier")
+    _base_doc = LGBMModel.fit.__doc__.replace("self : LGBMModel", "self : LGBMClassifier")  # type: ignore
```

Comment:
warnings about `.__doc__` being `Optional[str]` could be avoided by changing this to something like `getattr(LGBMModel.fit, "__doc__", "")`, but I think this is preferable, so that `.replace()` raises an exception in CI if something is accidentally changed that results in the docs being empty for a method.